server/datastore/mysql/scripts.go

```diff
@@ -129,6 +129,12 @@ func (ds *Datastore) SetHostScriptExecutionResult(ctx context.Context, result *f
     execution_id = ?`
 
 	const hostMDMActionsStmt = `
+  SELECT 'uninstall' AS action
+  FROM
+	host_software_installs
+  WHERE
+	execution_id = :execution_id AND host_id = :host_id
+  UNION -- host_mdm_actions query (and thus row in union) must be last to avoid #25144
```

Comment:
Was it getting `action` empty string before this fix?

Reply:
If a host has been locked/unlocked since the most recent enrollment, it will have a row in `host_mdm_actions`. That row will evaluate as `''` from the query here, and rows are returned in the order the queries are `UNION`ed so if the MDM actions row exists it will always show up first.

There were a few other ways to solve this (see [this comment](https://github.com/fleetdm/fleet/issues/25144#issuecomment-2571783405)) but this one wound up being simplest.